spock-core/src/main/java/org/spockframework/mock/runtime/DynamicProxyMockInterceptorAdapter.java

```diff
@@ -25,6 +27,34 @@ public DynamicProxyMockInterceptorAdapter(IProxyBasedMockInterceptor interceptor
   }
 
   public Object invoke(Object target, Method method, Object[] arguments) throws Throwable {
+    boolean isDefault = (method.getModifiers() & (Modifier.ABSTRACT | Modifier.PUBLIC | Modifier.STATIC)) == Modifier.PUBLIC
```

Comment:
Ok, then it does what it should do, but it is easy to confuse what is intended and it should be clarified, e.g. by putting it in a appropriately named method.